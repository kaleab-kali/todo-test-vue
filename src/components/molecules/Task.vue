<template>
  <div>
    <div :class="$style.todo">
      <Checkbox @click="$emit('changes')" :isChecked="isChecked" />
      <span
        :class="[$style.task, { [$style.done]: isChecked }]"
        data-cy="task__title"
        >{{ title }}</span
      >
      <DeleteButton @click="$emit('delete')" />
    </div>
  </div>
</template>

<script>
import Checkbox from "@/components/atoms/CheckBox";
import DeleteButton from "@/components/atoms/DeleteButton";
export default {
  props: {
    title: String,
    isChecked: Boolean,
  },
  components: {
    Checkbox,
    DeleteButton,
  },
};
</script>

<style lang="scss" module>
.todo {
  font-family: "Inter";
  background-color: $darkPeach;
  max-width: 30rem;
  margin-bottom: 1.56rem;
  display: flex;
  color: $brown;
  font-weight: 400;
  font-size: 1.25rem;
  padding: 0.688rem 0;
  margin-top: 0;
  border-radius: 0.625rem;
  .done {
    text-decoration: line-through;
  }
  @media (max-width: 980px) {
    max-width: 30rem;
    margin: 1rem;
  }
  @media (max-width: 850px) {
    font-size: 1rem;
  }
  @media (max-width: 600px) {
    font-size: 0.8rem;
  }
  .task {
    margin-left: -8rem;
    position: relative;
    @media (max-width: 850px) {
      input {
        font-size: 1rem;
      }
    }
    @media (max-width: 600px) {
      margin-left: -6rem;
      input {
        max-width: 10rem;
        font-size: 0.9rem;
      }
    }
    @media (max-width: 450px) {
      margin-left: -3rem;
    }
  }
}
.todo:last-child {
  margin-bottom: 0.56rem;
}
</style>
