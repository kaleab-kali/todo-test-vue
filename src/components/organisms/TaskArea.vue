<template>
  <div :class="$style.taskArea">
    <div :class="$style.empty" v-if="filterTasks.length == 0" data-cy="notasks">
      The task list is empty! <br />
      Add a new task!
    </div>
    <div :class="$style.tasksContent">
      <transition-group>
        <Task
          v-for="todo in filterTasks"
          :key="todo.id"
          :id="todo.id"
          :title="todo.title"
          :isChecked="todo.isChecked"
          @delete="deleteTodo(todo.id)"
          @changes="changeTaskStatus(todo.id)"
        />
      </transition-group>
    </div>
    <AddTask @keyup="addTodoItem" />
  </div>
</template>

<script>
import { mapGetters, mapMutations } from "vuex";
import Task from "@/components/molecules/Task";
import AddTask from "@/components/atoms/AddTask";
export default {
  data() {
    return {
      value: "",
    };
  },
  methods: {
    ...mapMutations(["changeTaskStatus"]),
    deleteTodo(id) {
      this.$store.commit("deleteTask", id);
    },
    addTodoItem(value) {
      this.$store.commit("addTodo", value);
    },
    addSubTask() {},
  },
  computed: {
    ...mapGetters(["filterTasks"]),
  },
  components: {
    Task,
    AddTask,
  },
};
</script>

<style lang="scss" module>
.taskArea {
  background-color: $lightPeach;
  padding-top: 2.875rem;
  padding-bottom: 1.875rem;
  .tasksContent {
    min-height: 15rem;
    max-height: 22rem;
    overflow-y: auto;
    width: 97%;
  }
  ::-webkit-scrollbar {
    width: 0.5rem;
    border-radius: 5px;
    background-color: $lightOrange;
  }
  ::-webkit-scrollbar-thumb {
    border-radius: 5px;
    background-color: $backgroundGradient;
  }
  .empty {
    margin-bottom: -8rem;
    font-size: 2rem;
    font-family: "Open Sans";
    font-weight: 600;
    color: $lightBrown;
    text-align: center;
  }
  @media (max-width: 800px) {
    padding-top: 1rem;
  }
}
.add {
  font-family: "Inter";
  padding: 0.688rem 0;
  border-radius: 0.625rem;
  display: flex;
  color: $lightOrange;
  background-color: $peach;
  border: 2px dashed $lightOrange;
  max-width: 30rem;
  @media (max-width: 980px) {
    max-width: 30rem;
    margin: 1rem;
  }
  @media (max-width: 450px) {
    margin-bottom: 0;
  }
  input {
    font-weight: 400;
    font-size: 1.25rem;
    color: $lightOrange;
    background-color: $peach;
    border: none;
    display: flex;
    text-align: center;
    @media (max-width: 450px) {
      max-width: 10rem;
      font-size: 1rem;
    }
  }
}
</style>
