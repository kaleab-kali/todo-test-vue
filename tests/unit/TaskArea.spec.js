import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import TaskArea from "@/components/organisms/TaskArea.vue";
import Task from "@/components/molecules/Task.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Organism TaskArea Component", () => {
  let store;
  let mutations;
  let getters;

  beforeEach(() => {
    mutations = {
      addTodo: jest.fn(),
      deleteTask: jest.fn(),
    };
    getters = {
      filterTasks: () => jest.fn(),
    };
    store = new Vuex.Store({
      mutations,
      getters,
    });
  });
  it("Correct layout", () => {
    const wrapper = mount(TaskArea, {
      store,
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("calls mutation addTodo", async () => {
    const wrapper = mount(TaskArea, {
      store,
      localVue,
    });
    await wrapper.find("input").trigger("keyup.enter");
    expect(mutations.addTodo).toHaveBeenCalled();
  });
  it("calls getter filterTasks", () => {
    const wrapper = mount(TaskArea, { store, localVue });
    expect(wrapper.findComponent(Task)).toBeTruthy();
  });
});
