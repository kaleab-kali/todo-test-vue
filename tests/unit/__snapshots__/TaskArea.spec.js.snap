// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism TaskArea Component Correct layout 1`] = `
<div class="taskArea">
  <div class="add"><input type="text" placeholder="Add new task"></div>
</div>
`;

exports[`TaskArea.vue Correct layout 1`] = `
<div class="taskArea">
  <div class="add"><input type="text" placeholder="Add new task"></div>
</div>
`;
